crates/p2p_proto/src/common.rs

```diff
@@ -232,6 +235,36 @@ pub struct CommonTransactionReceiptProperties {
     #[optional]
     pub consumed_message: Option<MessageToL2>,
     pub execution_resources: ExecutionResources,
+    pub execution_status: ExecutionStatus,
+    pub revert_error: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Dummy)]
+pub enum ExecutionStatus {
+    Succeeded,
+    Reverted,
+}
+
+impl TryFromProtobuf<i32> for ExecutionStatus {
+    fn try_from_protobuf(input: i32, field_name: &'static str) -> Result<Self, std::io::Error> {
+        match input {
+            0 => Ok(ExecutionStatus::Succeeded),
+            1 => Ok(ExecutionStatus::Reverted),
```

Comment:
Since this number -> enum value association is part of the Protobuf definition you shouldn't do this manually:  `proto::common::ExecutionStatus::from_i32()` should be used to parse the integer.

Reply:
That's only true if the mapping is consistent between the implementations though? No guarantee that that's the case if you want to reason locally? Or maybe I'm misunderstanding

Reply:
Oh wait I think I understand.

Reply:
Actually I don't understand why we have these extra types -- shouldn't all the types be defined in the proto files and autogenerated?

Reply:
We actually _could_ use the generated enums here if we want. However, since all fields in a message are optional using the generated structs for messages would be _really_ painful.

Reply:
So using just the generated types in general on our p2p API wouldn't really work and I think special-casing _some_ types (like enums which we don't have too  many of) doesn't make sense.